codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrap.java

```diff
@@ -137,13 +130,37 @@ public void run() {
                     }
                 });
             }
+        } catch (Throwable cause) {
+            promise.setFailure(cause);
         }
         return promise;
     }
 
+    private ChannelHandlerContext findCtx() throws ClosedChannelException {
+        // First try to use cached context and if this not work lets try to lookup the context.
+        ChannelHandlerContext ctx = this.multiplexCtx;
+        if (ctx != null && !ctx.isRemoved()) {
+            return ctx;
+        }
+        ctx = channel.pipeline().context(Http2MultiplexCodec.class);
```

Comment:
2 calls to `channel.pipeline()` could be moved to a variable

Reply:
Sure I can do... that said I think it not really matters

Reply:
@doom369 addressed... please re-review.